front/pages/api/w/[wId]/groups/[gId]/index.ts

```diff
@@ -0,0 +1,110 @@
+import type { GroupType, UserType, WithAPIErrorResponse } from "@dust-tt/types";
+import { PatchGroupRequestBodySchema } from "@dust-tt/types";
+import { isLeft } from "fp-ts/lib/Either";
+import * as reporter from "io-ts-reporters";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { GroupResource } from "@app/lib/resources/group_resource";
+import { UserResource } from "@app/lib/resources/user_resource";
+import { apiError } from "@app/logger/withlogging";
+
+export type GetGroupResponseBody = {
+  group: GroupType & {
+    members: UserType[];
+  };
+};
+
+export type PatchGroupResponseBody = {
+  group: GroupType;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<
+    WithAPIErrorResponse<GetGroupResponseBody | PatchGroupResponseBody>
+  >,
+  auth: Authenticator
+): Promise<void> {
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you requested was not found.",
+      },
+    });
+  }
+
+  const group = await GroupResource.fetchById(auth, req.query.gId as string);
```

Comment:
`nit` but we avoid a cast:

```suggestion
  const {gId} = req.query;
  
  if (typeof gId !== "string") {
    return apiError(req, res, {
      status_code: 404,
      api_error: {
        type: "group_not_found",
        message: "The group you requested was not found.",
      },
    });
  }
  
  const group = await GroupResource.fetchById(auth, gId);
```